src/api/core/queries/filterableQuery.ts

```diff
@@ -25,11 +25,7 @@ export abstract class FilterableQuery<T> extends BaseQuery<T> {
    */
   public where(
     filter: IFilteringCriterion<T> | IFilteringCriterionCallback<T>): this {
-    this.query.setFilterCriteria(
-      typeof filter === "function" ?
-        filter((field) => new FieldFilter(field)) :
-        filter,
-    );
+    this.query.setFilterCriteria(toFilteringCriterion(filter));
```

Comment:
I would move `toFileringCriterion()` logic inside `Query` class. Would be nice to have just:
```
this.query.setFilterCriteria(filter);
```
wouldn't it?

Reply:
yes, sir